src/Column/DataColumn.php

```diff
@@ -0,0 +1,427 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\DataView\Column;
+
+use Closure;
+use InvalidArgumentException;
+use Stringable;
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Html\Html;
+use Yiisoft\Html\Tag\Input;
+use Yiisoft\Html\Tag\Select;
+use Yiisoft\Yii\DataView\Helper\Attribute;
+
+use function sprintf;
+
+/**
+ * DataColumn is the default column type for the {@see GridView} widget.
+ *
+ * It is used to show data columns and allows {@see sortingEnabled|sorting} and {@see filter|filtering} them.
+ *
+ * A simple data column definition refers to an attribute in the data of the GridView's data provider.
+ *
+ * The name of the attribute is specified by {@see attribute}.
+ *
+ * By setting {@see value} and {@see label}, the label and cell content can be customized.
+ *
+ * A data column differentiates between the {@see getDataCellValue|data cell value} and the
+ * {@see renderDataCellContent|data cell content}. The cell value is an un-formatted value that may be used for
+ * calculation, while the actual cell content is a {@see format|formatted} version of that value which may contain HTML
+ * markup.
+ */
+final class DataColumn extends Column
+{
+    private string $attribute = '';
+    private string $filter = '';
+    private string $filterAttribute = '';
+    private array $filterInputAttributes = [];
+    /** @psalm-var string[] */
+    private array $filterInputSelectItems = [];
+    private string $filterInputSelectPrompt = '';
+    private string $filterModelName = '';
+    private string $filterType = 'text';
+    /** @psalm-var string[] */
+    private array $filterTypes = [
```

Comment:
Can I add my own type?

Reply:
What other types are needed, there are the most used with the input tag, if you want to add a custom filter, you can do it with the `filter()` method.